src/components/Cart/Cart.js

```diff
@@ -0,0 +1,133 @@
+import React, { useContext } from "react";
+import { formatMoney } from "../../utils/formatUtils";
+import { Link } from "react-router-dom";
+import { ShoppingCartContext } from "../../utils/context/ShoppingCartContext";
+
+const Cart = () => {
+  const [shoppingCart, setShoppingCart] = useContext(ShoppingCartContext);
+  const isShoppingCartEmpty = shoppingCart.count === 0;
+
+  const deleteCartItem = (cartItemId) => {
+    const cartItem = shoppingCart.cartInfo.find(
+      (cartItem) => cartItem.id == cartItemId
+    );
+    const copyCartInfo = shoppingCart.cartInfo.filter(
+      (cartItem) => cartItem.id != cartItemId
+    );
+
+    setShoppingCart((prevShoppingCart) => ({
+      count: prevShoppingCart.count - parseInt(cartItem.quantity),
+      cartInfo: copyCartInfo,
+    }));
+  };
+
+  const minusCartItem = (cartItemId) => {
+    let copyCartInfo = [...shoppingCart.cartInfo];
+    const indexItem = shoppingCart.cartInfo.findIndex(
+      (cartItem) => cartItem.id == cartItemId
+    );
+
+    if (copyCartInfo[indexItem].quantity > 1) {
+      copyCartInfo[indexItem].quantity -= 1;
+    } else {
+      copyCartInfo = copyCartInfo.filter(
+        (cartItem) => cartItem.id != cartItemId
+      );
+    }
+
+    setShoppingCart((prevShoppingCart) => ({
+      count: prevShoppingCart.count - 1,
+      cartInfo: copyCartInfo,
+    }));
+  };
+
+  const plusCartItem = (cartItemId) => {
+    const copyCartInfo = [...shoppingCart.cartInfo];
+    const indexItem = shoppingCart.cartInfo.findIndex(
+      (cartItem) => cartItem.id == cartItemId
```

Comment:
Moving this code to a useReducer, there is also code in ProductGridCard , that I am going to refactor